### PR TITLE
Allow excluding nodes from gql generation

### PR DESCRIPTION
**Pull request by cliedeman**
Allow users to omit types. This will allow users to onboard one entity at a time. 

Not very useful right now but the general skip annotation features into some planned work. 

Skipping fields for example.
This will allow users to add the gqlgen integraton incrementally

### CODE REVIEW

File: entgql/templatefuncs_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package entgql
+
+import (
+	"entgo.io/ent/entc/gen"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
```
Comment (a8m) [inline entgql/templatefuncs_test.go:21]:
```suggestion
import (
	"testing"

	"entgo.io/ent/entc/gen"

	"github.com/stretchr/testify/require"
)
```

File: entgql/templatefuncs_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			Annotations: map[string]interface{}{
+				annotationName: map[string]interface{}{},
+			},
+		},
+		{
+			Name:   "Type2",
+			Config: &gen.Config{},
+		},
+	}, nodes)
+}
```
Comment (a8m) [inline entgql/templatefuncs_test.go:56]: 👌

**Review by a8m — APPROVED** [review]
Thanks for adding this @cliedeman! 

Please fix these small style changes and I'll merge this.

File: entgql/annotation.go [hunk 3]
```diff
@@ -14,17 +14,22 @@
...
+import (
+	"encoding/json"
+	"entgo.io/ent/schema"
+)
 
 // Annotation annotates fields and edges with metadata for templates.
 type Annotation struct {
 	// OrderField is the ordering field as defined in graphql schema.
-	OrderField string
+	OrderField string `json:"OrderField,omitempty"`
```
[the remark this answers is not in the document]
Comment (cliedeman) [reply entgql/annotation.go:25]: happy to make the change. Just a note that all the entgql templates will have to change

[the remark this answers is not in the document]
Comment (a8m) [reply entgql/annotation.go:25]: Got it. So please ignore this suggestion

**Review by a8m — APPROVED** [review]
Looks good @cliedeman 😃 

Thanks for adding this 🙏

**Pull request merged**